Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -837,7 +837,7 @@ public void onError(final RequestError error) {
       final boolean ok = runInPooledThreadAndWait(new Runnable() {
         @Override
         public void run() {
-          server.edit_format(filePath, selectionOffset, selectionLength, consumer);
+          server.edit_format(filePath, selectionOffset, selectionLength, -1, consumer);
```

Comment:
I think this is correct parameter. Please check that it will work when you change margin in Preferences | Editor | Code Style | Dart | Wrappijng and Braces | Right margin.
CodeStyleSettingsManager.getSettings(project).getCommonSettings(DartLanguage.INSTANCE).RIGHT_MARGIN